activerecord/lib/active_record/reflection.rb

```diff
@@ -437,7 +437,8 @@ def compute_class(name)
           end
 
           klass
-        rescue NameError
+        rescue NameError => e
+          msg = [msg, "The original error was: #{e.message}"].join("\n")
```

Comment:
I think ideally we'd work out whether `e` is complaining about the name we "expect", and probably let the original exception through untouched if not?

(While we're looking here, I don't love the "If" I see above in the error message -- is it practical for us to figure that out and only suggest it when applicable?)

Reply:
It's also a bit redundant with `#cause` (which may or may not be displayed depending on what shows you the error).

Makes me wonder if the Rails error page shows the error cause 🤔

Reply:
Yes, it does. But last time I investigated this problem the cause wasn't being set. Maybe we should just do that.

Reply:
I'm going to close this for now because when I tried to replicate the issue the cause [was set](https://github.com/rails/rails/issues/46532#issuecomment-1329878733) (though I'm not sure where...).

Reply:
yeah, if you raise from inside a `rescue` block, the cause is automatically set.